editoast/src/views/timetable/track_occupancy.rs

```diff
@@ -175,69 +209,105 @@ pub fn find_track_occupancy_for_operational_point(
+                .schedule
+                .iter()
+                .find(|schedule| schedule.at.0 == train_schedule.path[matching_index].id.0);
+            let arrival_time = report_train.path_item_times[matching_index] as i64;
+            let stop_duration = schedule_item
+                .and_then(|item| item.stop_for.clone())
+                .unwrap_or_default();
+
+            if let Some(track_section) = get_track_section_from_position(
+                pathfinding_result_success.path_item_positions[matching_index],
+                &path_projection,
+                operational_point_track_offsets,
+            ) {
+                return vec![TrackOccupancy {
+                    track_section,
+                    time_window: TimeWindow {
+                        time_begin: train_schedule.start_time
+                            + Duration::milliseconds(arrival_time),
+                        duration: stop_duration,
+                    },
+                }];
+            }
+        } else {
+            // No direct match: interpolate from report_train
+            return interpolate_track_occupancy(
+                train_schedule,
+                &path_projection,
+                operational_point_track_offsets,
+                report_train,
             );
-            (
-                &pathfinding_result_success.path.track_section_ranges,
-                &pathfinding_result_success.path_item_positions,
-            )
-        }
-        PathfindingResult::Failure(_) => {
-            // The timetable can have a pathfinding failure for a train schedule
-            // because the train schedule is not valid.
-            // We skip it because we don't want to compute the track occupancy for an invalid train schedule.
-            // It's not a problem because we are just looking for the track occupancy of the operational point for the valid trains.
-            tracing::info!(train_schedule.train_name, "pathfinding failed");
-            return vec![];
         }
-    };
-    let path_projection = PathProjection::new(track_ranges);
+    }
+    //Invalid trains
+    else {
+        let pathfinding_success = match pathfinding {
+            PathfindingResult::Success(success) => Some(success),
+            _ => None,
+        };
+        let (scheduled_path, path_item_positions) =
+            build_scheduled_path_and_positions(train_schedule, pathfinding_success);
 
-    // Get the positions and the times from the simulation
-    let report_train = match simulation {
-        simulation::Response::Success(SimulationResponseSuccess { final_output, .. }) => {
-            &final_output.report_train
-        }
-        _ => {
-            tracing::info!(train_schedule.train_name, "simulation failed");
-            return vec![];
+        if !scheduled_path.is_empty()
+            && let Some(matching_index) =
+                find_matching_path_item(&scheduled_path, operational_point_id, path_item_cache)
```

Comment:
We don't need to check if `scheduled_path` is empty
```suggestion
        if let Some(matching_index) =
                find_matching_path_item(&scheduled_path, operational_point_id, path_item_cache)
```

Reply:
Done